react/week3/todo-app/src/AddTodo.js

```diff
@@ -0,0 +1,111 @@
+import React, { useState, useEffect } from "react";
+import { getNextId, getTodos } from "./TodosHelper";
+import Todo from "./Todo";
+
+const AddTodo = function () {
+  const [list, setList] = useState([]);
+  const [textValue, changeTextValue] = useState("");
+  const [dateValue, changeDateValue] = useState("");
+  const [editMode, updateEditMode] = useState(-1);
+  const [newDescription, changeDescription] = useState("");
+
+  useEffect(() => {
+    setList([]);
+    getTodos().then((data) => setList(data));
+  }, []);
+
+  const nextId = getNextId(list);
+  let newTodo = { id: nextId, description: textValue, deadline: dateValue };
```

Comment:
I'd suggest you add these inside the addItem function, you don't really need to use them anywhere else so it's good to keep it together :)